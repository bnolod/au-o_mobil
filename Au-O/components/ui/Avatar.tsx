import { AvatarProps } from "@/constants/types";
import { Image, View } from "react-native";
import ThemedText from "./ThemedText";

export default function Avatar({
  image,
  className,
  nickname,
  height = 14,
  width = 14,
}: AvatarProps) {
  if (image) {
    return (
      <Image
        source={{ uri: image }}
        className={`w-${width} h-${height} ${className} rounded-full`}
      />
    );
  } else if (!image && nickname) {
    return (
      <View
        className={`${className} ${
          className
            ? className
            : "bg-backdrop-secondary dark:bg-backdrop-secondary-dark"
        } w-${width} h-${height} flex items-center justify-center text-center rounded-full`}
      >
        <ThemedText className="text-lg font-black">
          {nickname[0].toUpperCase()}
          {nickname[1].toUpperCase()}
          {nickname[2].toUpperCase()}
        </ThemedText>
      </View>
    );
  } else if (!image && !nickname) {
    return (
      <View
        className={
          className
            ? className
            : " bg-backdrop-secondary dark:bg-backdrop-secondary-dark w-14 h-14 flex items-center justify-center text-center rounded-full"
        }
      >
        <ThemedText className="text-lg font-black">???</ThemedText>
      </View>
    );
  }
}
