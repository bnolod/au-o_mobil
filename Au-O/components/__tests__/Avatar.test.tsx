import * as React from 'react';
import {render} from '@testing-library/react-native';
import Avatar from '@/components/ui/Avatar';

test('Avatar komponens létezik', () => {
    expect(Avatar).toBeDefined()
})

test('Avatar fallback szöveg megjelenik', () => {
    const {getByText} = render(<Avatar image={null} nickname='TESZT'/>)

    expect(getByText("TES")).toBeDefined()
})
test('Avatar fallback szöveg nélkül is megjelenik', () => {
    const {getByText} = render(<Avatar image={null} nickname={undefined}/>)

    expect(getByText("???")).toBeDefined()
})
test('Avatar kép megjelenik', () => {
    const tree = render(<Avatar image='https://via.placeholder.com/150' nickname='TESZT'/>)

    expect(tree).toMatchSnapshot()
})