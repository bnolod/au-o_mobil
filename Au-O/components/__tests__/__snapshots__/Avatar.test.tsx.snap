// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar k√©p megjelenik 1`] = `
<Image
  className="w-14 h-14 undefined rounded-full"
  source={
    {
      "uri": "https://via.placeholder.com/150",
    }
  }
/>
`;
